can/interfaces/vector/canlib.py

```diff
@@ -93,6 +97,24 @@ def __init__(self, channel, can_filters=None, poll_interval=0.01,
         # Get channels masks
         self.channel_masks = {}
         self.index_to_channel = {}
+
+        if serial is not None:
+            app_name = None
+            channel_index = []
+            channel_configs, channel_count = get_channel_configs()
+            for i in range(channel_count):
+                channel_config = channel_configs[i]
+                if channel_config.serialNumber == serial:
+                    if channel_config.hwChannel in self.channels:
+                        channel_index.append(channel_config.channelIndex)
+            if len(channel_index) > 0:
+                if len(channel_index) != len(self.channels):
+                    LOG.info("At least one defined channel wasn't found on the specified hardware.")
+                self.channels = channel_index
+            else:
+                # Is there any better way to raise the error?
+                raise Exception("None of the configured channels could be found on the specified hardware.")
```

Comment:
Maybe `ValueError`? Should be changed in #356 anyway.

Reply:
So may I leave it as it is for now?

Reply:
Or change to `CanError` as that is what most other interfaces will raise (or a subclass of it).